src/compiler/scala/tools/nsc/javac/JavaParsers.scala

```diff
@@ -326,20 +336,55 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
     }
 
     def annotations(): List[Tree] = {
-      //var annots = new ListBuffer[Tree]
+      var annots = new ListBuffer[Tree]
       while (in.token == AT) {
         in.nextToken()
-        annotation()
+        annots += annotation()
       }
-      List() // don't pass on annotations for now
+      annots.toList
     }
 
     /** Annotation ::= TypeName [`(` AnnotationArgument {`,` AnnotationArgument} `)`]
      */
-    def annotation() {
-      qualId()
-      if (in.token == LPAREN) { skipAhead(); accept(RPAREN) }
-      else if (in.token == LBRACE) { skipAhead(); accept(RBRACE) }
+    def annotation(): Tree = {
+      def annArg(): Tree = {
+        def annVal(): Tree = {
```

Comment:
Hmm. So, I can easily skip the value until the next comma (modulo parens). But... in that case, what would you expect the value to be? In the bit where we check to see if we can give a `static final` field a `ConstantType`, we give up and return the declared type if it's not a literal (which makes sense). 

However, I'm not sure what `Tree` we could give them. `EmptyTree`?
 `TypeTree().setType(whateverType)`? Someone will be surprised.

Should there be a warning? It seems possibly cruel to people who use `-Xfatal-warnings`, but since it should only affect Java source which the user can change, it's not the end of the world. If there were a way to make a `Tree` which warns the first time you look at it... but that's silly.

I could file a bug about that and label it "notatallquickfix", and we could call this "known restriction".

What do you think?